src/lib/wallet/MultipleAddressWallet.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import HDKey from 'hdkey'
+import Wallet from 'ethereumjs-wallet'
+import logger from '../logger/pino-logger'
+
+
+const log = logger.child({ from: 'MultipleAddressWallet' })
+
+type WalletsCollection = {
+  [key: string]: Wallet // Associative array
+}
+class MultipleAddressWallet {
+  ready: Promise<Web3>
+  wallet: Wallet
+  wallets: WalletsCollection
+  mnemonic: string
+  addresses: Array<string>
+  numOfAccounts: number = 10
+
+  constructor(mnemonic: string, numOfAccounts: number) {
+    logger.debug('MultipleAddressWallet ', { mnemonic }, { numOfAccounts })
+    this.numOfAccounts = numOfAccounts
+    this.mnemonic = mnemonic
+    this.addresses = []
+    this.wallets = {}
+  }
+
+  async ready() {
+    await this.initAccounts()
+    this.wallet = this.wallets[0]
+  }
+
+  async initAccounts() {
```

Comment:
It's defined async but await is not used, so i'm not sure when awaiting on initaccounts it actually waits until accounts have been created

Reply:
it doesn't look like an async code. Probably triggering `initAccount` in the constructor will do the trick.

Reply:
yes but also no await is used inside initaccounts and no promises are returned so even if we await on initaccounts it won't have the expected results

Reply:
I was pointing that where it's used:

```javascript
// ...
    let mulWallet = new MultipleAddressWallet(pkey, 10)
    await mulWallet.ready()
    let web3 = new Web3(provider)
// ...
```

we can get rid of the `await mulWallet.ready()` line and of any `await/then` sentence, as `MultipleAddressWallet` initialization is synchronous.

Reply:
done. thanks